platform/viewer/src/components/ViewportGrid.jsx

```diff
@@ -43,30 +48,56 @@ function ViewerViewportGrid(props) {
     };
   }, []);
 
-  // TODO -> Make a HangingProtocolService
-  const HangingProtocolService = displaySets => {
-    let displaySetInstanceUID;
-
-    // Fallback
-    if (!displaySets || !displaySets.length) {
-      const displaySet = DisplaySetService.activeDisplaySets[0];
-      displaySetInstanceUID = displaySet.displaySetInstanceUID;
-    } else {
-      const displaySet = displaySets[0];
-      displaySetInstanceUID = displaySet.displaySetInstanceUID;
-    }
+  useEffect(() => {
+    const { unsubscribe } = MeasurementService.subscribe(
+      MeasurementService.EVENTS.JUMP_TO_MEASUREMENT,
```

Comment:
I'm torn between this happening:

1. In the `ViewportGrid` component
2. In the `ViewportGridService`
3. In the `Viewport` extension module component

I'm leaning toward number 3, then mixed on 1 & 2.

For 3, we already have the subscribe, and the logic's in one place, and the viewport could handle any wonkiness if it needs alternative behavior.

For 1, it's another responsibility the Viewer project has, when we're trying to make it _as minimal as possible_ as it becomes part of the un-reusable glue if you want to provide an alternative `Viewer` project.

For 2, it's more reusable for those that want to create their own `Viewer` project, but creates a more direct service to service dep.

Reply:
I'm against 2, as that links the two services together expliclty.

3 we have to write for _every_ viewport, and it will be the same behavior. The Measurement references a displaySet, so if that displaySet is not present, it will always be changed/

Hence why I went with 1, happy to discuss.

Reply:
I think I'm fine with it. I'm just trying to figure out _what_ things like this are, and how we document it as a requirement of the "glue" project; something that's not solved by the reusable part of the platform.